src/java/io/bazel/rulesscala/specs2/Specs2RunnerBuilder.scala

```diff
@@ -0,0 +1,52 @@
+package io.bazel.rulesscala.specs2
+
+import java.util.regex.Pattern
+
+import io.bazel.rulesscala.test_discovery.{FilteredRunnerBuilder, PrefixSuffixTestDiscoveringSuite}
+import org.junit.runner.notification.RunNotifier
+import org.junit.runner.{Description, RunWith, Runner}
+import org.junit.runners.Suite
+import org.junit.runners.model.RunnerBuilder
+import org.specs2.control.Action
+import org.specs2.main.{Arguments, CommandLine, Select}
+import org.specs2.specification.core.Env
+import org.specs2.specification.process.Stats
+
+import scala.collection.JavaConverters._
+
+@RunWith(classOf[Specs2PrefixSuffixTestDiscoveringSuite])
+class Specs2DiscoveredTestSuite
+
+class Specs2PrefixSuffixTestDiscoveringSuite(testClass: Class[Any], builder: RunnerBuilder)
+  extends Suite(new Specs2RunnerBuilder(builder), PrefixSuffixTestDiscoveringSuite.discoverClasses())
+
+class Specs2RunnerBuilder(builder: RunnerBuilder) extends FilteredRunnerBuilder(builder) {
```

Comment:
In the previous commit, this class was internal to the `FilteredRunnerBuilder`.
What is the motivation for this change (switch to inheritance)?
It makes the code a little harder to follow...

Reply:
I made this change together with @ittaiz, in order to decouple specs2-specific stuff from the `FilteredRunnerBuilder`. This inheritance allows using specs2-related dependencies only in case of a specs2 test (when using the `scala_specs2_junit_test` rule directly)

Reply:
I guess there are other ways to achieve this decoupling.

How about introducing a delegating builder that identifies which test is being run and delegates the filtering to the correct builder (junit or specs2 in this case).

WDYT?

Reply:
better decoupling